lib/std/crypto/gimli.zig

```diff
@@ -94,7 +95,57 @@ pub const State = struct {
         }
     }
 
-    pub const permute = if (std.builtin.mode == .ReleaseSmall) permute_small else permute_unrolled;
+    const Lane = Vector(4, u32);
+
+    inline fn shift(x: Lane, comptime n: comptime_int) Lane {
+        return x << @splat(4, @as(u5, n));
+    }
+
+    inline fn rot(x: Lane, comptime n: comptime_int) Lane {
+        return (x << @splat(4, @as(u5, n))) | (x >> @splat(4, @as(u5, 32 - n)));
+    }
+
+    fn permute_vectorized(self: *Self) void {
+        const state = &self.data;
+        var x = Lane{ state[0], state[1], state[2], state[3] };
```

Comment:
Would `@bitCast(Lane, state[0..4].*)` be a better choice?

Reply:
This is fine on Intel and ARM, but I'm not too comfortable about making general assertions on alignment and ordering compatibility between arrays and vectors. Since the compiler optimises that version to a single load, and it is simpler, I'd rather stick to it :)